### DISCUSSION TITLE
Ideas incubator

### DISCUSSION THREAD

**Discussion by ollesmaps** [OpenOrienteering/mapper#1035] [Ideas]
I miss the opportunity to incubate ideas somewhere, to talk about OOMapper new possible functions or ask some silly questions others might find obvious. Could that be here? And I mean here as in comments bellow this issue as an ever expanding conversation.

**Comment by dg0yt** [comment]
Discussion is meant to happen here. 

 - There are 32 people "watching" this project, so I assume they get notified even without being mentioned explicitly.
 - Please try to keep different ideas in separate issues. (Cross-referencing is  easy, and this is another reason why discussion works well here.)
 - Please take the time look for existing issues.

For mere questions like this one, I attach the label 'question'.

**Comment by ollesmaps** [comment]
Committing, Pushing and Pulling does it relate to code only? Or could you commit let's say to improving visual style of icons? If there is a good explanation of Github, please point me there.

(Sample question, would that be really for a separate issue?)

**Comment by dg0yt** [comment]
Everything file in the repository is subject to committing, pushing and pulling. Of course, features such as displaying "diffs" work much better with text files.

I don't know about a single document in [Github Help](https://help.github.com/). You will have to learn how to 
- create your "fork" of our git repository
- create and maintain "branches" within your fork
- finally create "pull requests" for incorporating your changes into our repository.

Partly this can be managed even in the Github website, but there are limits. In order to work with a local clone of the repositories on your computer, its probably most convenient for non-developers to use the [Github desktop software](https://desktop.github.com/).

**Comment by ollesmaps** [comment]
Once a map is opened in Anroid app, is there a way where to find map scale?
Once a map is opened on desktop, is the only place where to find map scale at Change map scale?

**Comment by lpechacek** [comment]
> Discussion is meant to happen here.

My personal preference is a mailing list. [openorienteering-devel at SourceForge](https://sourceforge.net/p/oorienteering/mailman/) still works and might IMHO serve as a good place for discussing future directions. Question remains what are the preferences of the other thirty watchers.

Ad the map scale display in Android version, current display scale (zoom * map scale) could perhaps be part of the information toast, like "2x, 1:7500".

**Comment by ollesmaps** [comment]
@lpechacek With regards to scale - I meant a defined scale for the whole file. You can not find it anywhere while using Android.

**Comment by ollesmaps** [comment]
Is there a way how to add dash points while drawing in app? Not after drawing is finished, during drawing on Android device?

**Comment by lpechacek** [comment]
@ollesmaps: In my proposal the map scale would be visible after switching to 1x zoom (new feature in v0.8). I understand that it feels hackish. On the other hand, the user interface is not polluted with more dialogs and/or buttons.

Ad dash points, I think it's the **Dash** button that appears after selecting curve drawing tool. It's the third button from right.

**Comment by ollesmaps** [comment]
@lpechacek This is a simple solution. I think that shall be enough.

I did not notice the Dash button. Thank you for pointing it out!

**Comment by Zerbembasqwibo** [comment]
Today we use maps in more and more digital applications. The contour is not defined on a map, what is up? With skills or slope lines we can decide. But a computer will fail without slope lines. Why not define contour in OOM to prepare for the future that is already here? I think immediately with autoproduced maps that you might add som feature that put slope lines in small pits drawn with contours. There must also be more ways to be used, let's guess if to draw cliffs with tags in some automised way.

Of course the program that produces the contuors must gt the possibility to define the slope of a contour, but that's another story.

**Comment by ollesmaps** [comment]
@Zerbembasqwibo You mean:  so Mapper is aware where is up and down and you can later for example create DEM?

**Comment by Zerbembasqwibo** [comment]
The use will be in future. I mean rather when you get auto countours from other tool, that tool might (soon?) present up and down-side of contours. Than OOM might implement a function: "Auto place slope lines on small pits" if line length is shorter than some limit. 

But yes, maybe you also might export info to DEM och 3D-tools for visualising, or hillshade.

Anyway, a very simple implemetation is to just do a countour symbol that has this distinction built in.

Bu today, feels strange that a program doesn't know what is up and what is down since all information is there when as las-data.

**Comment by dg0yt** [comment]
@Zerbembasqwibo Every contour has a left and a right side. You just need to define which side is up, and which is down. The biggest obstacle with left and right is that you will sometimes have flipped sign for one axis. In fact, this is even true for the y axis in Mapper, stored vs. displayed. But add the slope line as a dash symbol, and it will be quite clear to see (like it already is for earth bank or cliff). And even software might analyse the dash symbol, for determining up and down from the side which the dash symbol extends to. OTOH, you can as well just add an attribute with the absolute height to each contour.

Personally I'm inclined to really make the "slope line" symbol the contour's dash symbol. Why should the slope line move separately? Why shouldn't it simply take its direction from the contour?

**Comment by mlerjen** [comment]
@dg0yt: nice one!
![image](https://user-images.githubusercontent.com/15611653/37920923-94198840-3128-11e8-9dd6-93c382292bfc.png)

**Comment by dg0yt** [comment]
Yes, it is all there...

**Comment by Zerbembasqwibo** [comment]
Now we are talking! I like it too!

2018-03-26 19:06 GMT+02:00 Kai Pastor <notifications@github.com>:

**Comment by ollesmaps** [comment]
I use gpx also to track working hours - that is for field work.
Something to track hours off field (on tablet) - would be also useful.
If mapper could combine both to form a little statistics for each file that would be very useful.

Do you use any additional software to track working hours?
I would welcome this is built-in within Mapper, however something else might work for you?

**Comment by Zerbembasqwibo** [comment]
I use this page: http://www.mygpsfiles.com/app/

2018-04-18 8:36 GMT+02:00 Aleš Hejna <notifications@github.com>:

**Comment by ollesmaps** [comment]
Thank you @Zerbembasqwibo . That is a good site. 
I guess you can not download the final table with all the hours as xls or similar.

**Comment by Zerbembasqwibo** [comment]
Copy paste.

onsdag 18 april 2018 skrev Aleš Hejna <notifications@github.com>:

**Comment by ollesmaps** [comment]
Do you have a sentence you put on maps as a thank you note for the existence and effort put in Open Orienteering Mapper?

**Comment by Zerbembasqwibo** [comment]
It's a fast way to draw with freehand lines in field. Later if you want to change it to curves I think it would be great if the tranform takes place in some steps:

1. Choose free hand line object
2. Click Change to curves
3. Now you are in a transfer mode: both original and curved line is shown
4. Adjust manually if necessary
5. Ready, the old one deletes

As for now I copy paste but the line ends up in another position. Very often I want to adjust the curved line.

Or are there better/other ways of how to do this?

**Comment by dg0yt** [comment]
> Do you have a sentence you put on maps as a thank you note for the existence and effort put in Open Orienteering Mapper?

I guess the OpenOrienteering logo in our symbol sets would work very well.

**Comment by dg0yt** [comment]
> As for now I copy paste but the line ends up in another position. Very often I want to adjust the curved line.
> Or are there better/other ways of how to do this?

At least, there is the "Duplicate" tool (shortcut: `D`). It operates in-place.

**Comment by Zerbembasqwibo** [comment]
Thank you dg0yt!

**Comment by ollesmaps** [comment]
@dg0yt I made the app/ program icon into an .oom symbol as an alternative. It does not glitter that much, but still looks quite good. 

[OOM_icon.zip](https://github.com/OpenOrienteering/mapper/files/2016453/OOM_icon.zip)

**Comment by ghost** [comment]
> I made the app/ program icon into an .oom symbol as an alternative

@ollesmaps, Look nice, but still need improvements.

![screen 1](http://i.imgur.com/vnZlVvu.png)
![screen 2](http://i.imgur.com/2VN6SIZ.png)

As for me, we need new redesigned Mapper icon before v1.0 release. Current square icon look ugly and "OOM" abbreviation on current icon mostly associated with "out-of-memory" & "OOM Killer"...

- https://en.wikipedia.org/wiki/Out_of_memory
- https://en.wikipedia.org/wiki/OOM_Killer

**Comment by ollesmaps** [comment]
@Symbian9 Yes, it is not perfect, it can be improved upon.

**Comment by jmacura** [comment]
Maybe it's time to rework the OOM logo completely..? I would propose something simpler, like the attached (just a quick sketch of mine). I believe that the "OOM" text is unnecessary.

![oom-new-sketch-128](https://user-images.githubusercontent.com/5598693/40284292-5021f0c4-5c8d-11e8-9f42-533cb5751a2d.png)

**Comment by ollesmaps** [comment]
@jmacura I would rather use stylus instead of pen. That is what I found rather old style on the current OOM icon (which I like otherwise). OMapper replaced pencils :)
I assume if there is one day a new GUI. A new logo icon/ logo can be part of that change.

**Comment by jmacura** [comment]
@ollesmaps I would definitely vote for a stylus, but I have found my graphic skills too lame for drawing it aptly, hence the "old-school" pencil ;-)

**Comment by ghost** [comment]
As you remember, few years ago "OpenOrienteering" project migrated from SourceForge to Github...

- http://helb.github.io/goodbye-sourceforge/

..., and now GitHub sold to Microsoft :-(

- https://www.bloomberg.com/news/articles/2018-06-03/microsoft-is-said-to-have-agreed-to-acquire-coding-site-github

Where "OpenOrienteering" will migrate now? On GitLab?

- https://docs.gitlab.com/ee/user/project/import/github.html

**Comment by sikmir** [comment]
> Where "OpenOrienteering" will migrate now? On GitLab?

@Symbian9 What for? What is wrong with GH after acquiring it by MS? MS isn't going to close GH for open source projects so far. SF is another story.

**Comment by lpechacek** [comment]
>> Do you have a sentence you put on maps as a thank you note for the existence and effort put in Open Orienteering Mapper?

>  I guess the OpenOrienteering logo in our symbol sets would work very well.

Mapper logo appeared on a [WRE sprint map](https://photos.google.com/share/AF1QipOlMULgI63RUspzN9W7psm3Adn83i9OIRJA7oTtQqpiQYoAH9uV1gXfIJWk5l_jZQ/photo/AF1QipN0gqtkmd2TiAf8-q4GEVNugeJk8nMqdPsTtUIO?key=dFZVLVdLLWFwREFJSnZ0bzg4cVR3MFliOERwQVF3).

**Comment by ghost** [comment]
> Mapper logo appeared on a WRE sprint map.

@lpechacek, it is Mapper icon, not logo.

**Comment by ollesmaps** [comment]
I do not precisely know how reactions are counted. However you can make an ascending list based on reactions - ["popular issues"](https://github.com/OpenOrienteering/mapper/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

**Comment by Zerbembasqwibo** [comment]
EDIT: Any answers to the question why 416 is prioritised?

When in app on android,  in field, I select things. If there is a 416 nearby, why is that always chosen first? Even if it is not very close to where I try to select other things.

In the app on Android, I see all files in the "file" folder, also in subfolders. Since I store all templates there, it quick get a mess.  Also the order of files is something I don't understand, it's ordered on some parameter I cannot guess (not first letter, not date of latest saved file as would be "normal"). And could subfolders' content be hided?

Anyway, worked **126 hours in field with 0.8.1, and just one crash**! :D :D (Lack of memory I guess, since I by mistake tried to show a very big template.)

**Comment by jmacura** [comment]
I'd like to notify the holy developers of OOMapper about an important feature which is coming to an Android devices soon, in case they haven't met it yet.

New GNSS receiver chips installed inside ARM processors will allow for **carrier-phase measurements** of GNSS signal and a reading of a second carrier wave at the same time. Yay! I don't know how much is changing programmatically from the current state of affairs, but obviously something is changing, as they present a new API for these features. It seems desirable to have these features allowed in the Android version of OOMapper when they hit the market.

Some more info about it, can be listened here: https://youtu.be/vywGgSrGODU?t=22m40s or read in this transcript: http://gpsworld.com/how-to-achieve-1-meter-accuracy-in-android/
Some more info about the current support of dual frequency reading in devices is here: https://medium.com/@sjbarbeau/dual-frequency-gnss-on-android-devices-152b8826e1c

**Comment by ollesmaps** [comment]
Has anyone tried Mapper on Chrome OS? The [Pixel Slate tablets](https://www.theguardian.com/technology/2018/nov/27/google-pixel-slate-review-chrome-os-android-tablets) are looking good. They support also stylus.

**Comment by ghost** [comment]
WOW! *OCAD AG* added to `OCAD` new *"Bézier Curve Mode"* tool for draw paths that now has same behaviors as *"Draw paths"* tool from `OpenOrienteering Mapper`:

- http://ocad.com/blog/2018/12/bezier-curve-mode/

Here is page in Mapper Manual with "Draw paths" tool description: 

- https://www.openorienteering.org/mapper-manual/pages/toolbars.html

Look like "plagiarism" of `OpenOrienteering Mapper` UX by *OCAD AG* 😕

**Comment by dg0yt** [comment]
Mapper received a lot of positive feedback for this tool, and we should appreciate that OCAD AG implements this, too.

**Comment by dg0yt** [comment]
@Symbian9 
- Please stop citing my *opinion* on Facebook. It is obvious that I do not participate in discussions there.
- And to be more explicit: I consider it unfair to talk of plagiarism in this context. I very much respect the work of OCAD AG.
- And this is "Ideas incubator". Your post is off-topic, and it is not the first time. Please consider relevance before posting.

**Comment by ghost** [comment]
> stop citing my *opinion*

OK, just removed "plagiarism" meaning and your *opinion* from Facebook.

**Comment by mlerjen** [comment]
A question though.

As it is now, the templates get loaded at the upstart of the file (if switched on), respectively the moment switched to visible for the first time after upstart (if switched off at the beginning). After that the template is kept "on hold" if toggled (not reloaded after first load). True?

I could imagine two cases, where a reload of the template "if changed" could be useful.

1) In a team work having the partners .omap as a template and vice versa, both being connected by a cloud. The moment the partner would submit his updated file, my background template would be updated.

2) Creating actual templates live. f.eg. working with ultra-sonic measurement to detect vegetation position and density. The result could be transmitted to Mapper by an updated template.

So my question is, how big of an effort it would be to keep the used templates updated.

**Comment by lpechacek** [comment]
> ... a reload of the template "if changed" could be useful.

I concur. I can imagine a dialog saying that template file has changed on disk and whether the user wants to reload the template.

> So my question is, how big of an effort it would be to keep the used templates updated.

I think it is doable. Of course, someone will have to invest a piece of his/her life for the functionality to materialize.

Edit: as an example, this is the dialog Qt Creator (the tool I use for programming) displays when file changes on disk:
![image](https://user-images.githubusercontent.com/5584406/52122235-901bdf00-2622-11e9-90c4-766ee765fc51.png)

**Comment by lpechacek** [comment]
[discussion from PR#1203 ([RFC] Resize multiple objects at once)]

> [ollesmaps commented 3 days ago](https://github.com/OpenOrienteering/mapper/pull/1203#issuecomment-470930009) 
> Would be possible to use this tool also as "Move parallel"?
> Which can be used like this (staying with your @lpechacek   map sample) : snapping along the roads with fence symbol and then move  parallel inside to get it further from the road. Possible?

To a certain degree, depending on shape of the object in question.

However, my understanding is that the [Edit Line Tool](https://www.openorienteering.org/mapper-manual/pages/toolbars.html#tool_edit_line) was intended for moving lines in perpendicular direction. I'm successfully using it for ensuring minimum size gaps between fences and paths. The only tool limitation is that the it moves exactly one line segment at a time. Extending it to "move all segments" would perhaps bring the "move parallel" functionality. Opinions?

**Comment by mlerjen** [comment]
> However, my understanding is that the [Edit Line Tool](https://www.openorienteering.org/mapper-manual/pages/toolbars.html#tool_edit_line) was intended for moving lines in perpendicular direction. I'm successfully using it for ensuring minimum size gaps between fences and paths. The only tool limitation is that the it moves exactly one line segment at a time. Extending it to "move all segments" would perhaps bring the "move parallel" functionality. Opinions?

#650  Concentric shift would be very useful imo

**Comment by lpechacek** [comment]
Agree. Seeing that no one touched the issue for about two year, I've picked it up. See if I find time to tackle this interesting topic.

**Comment by ollesmaps** [comment]
Concentric shift would be very usefull :+1:  

@lpechacek  "I'm successfully using it for ensuring minimum size gaps between fences and paths."
But you adjust the minimum gap only "by eye", right? Or is there a minimum gap tool, I miss?

**Comment by mlerjen** [comment]
@ollesmaps: The 0.15 minidot for ISSOM... but you do not refer to such of course.
![image](https://user-images.githubusercontent.com/15611653/54119395-817cd080-43f5-11e9-880d-6222365d0620.png)

**Comment by lpechacek** [comment]
@ollesmaps, just like Martin, I'm using a special **helper symbol** to measure gaps. Another approach is adding "aura" to point symbols which covers the symbol extent + minimum gap (explained elsewhere in this GitHub repo).

![image](https://user-images.githubusercontent.com/5584406/54130681-1f31c900-4411-11e9-8f70-604c7a028275.png)

**Comment by mlerjen** [comment]
@ollesmaps: as for the aura: define a white color at the very bottom of the color table to use for the aura. Thus the arua will only be visible while the object is selected.
![image](https://user-images.githubusercontent.com/15611653/54131371-800dd100-4412-11e9-89df-54044708e407.png)

**Comment by ollesmaps** [comment]
Thank you @lpechacek, @mlerjen !

**Comment by Zerbembasqwibo** [comment]
What about rotating of a map to align to north? Will there be a better solution than to rotate all points, as far I see when you mix OOM, OCAD, Condes and Purlpe pen, you get it messy (at least I do...). Are there any chances doing it like usual CAD, ie a frame that you can place in any direction before you export to OCAD-format? Will it help?

**Comment by ghost** [comment]
> What about rotating of a map to align to north? Will there be a better solution than to rotate all points

FTR, According ISOM specifications map should be printed on paper, where North is on top of paper.

Mapper is Desktop-Publishing Software, oriented for creating & preparing for printing orienteering maps according ISOM specifications

So rotating map could be additional feature, but not as replacement to symbol rotation

**Comment by Zerbembasqwibo** [comment]
You are right, but today "north alignment" does only work if you use "up
aligned" north lines (in older ocad versions, I don't know newest) so how
does it work while switching between the programs?


måndag 1 april 2019 skrev app4soft <notifications@github.com>:

**Comment by dg0yt** [comment]
You have to draw the map as it is meant to be printed. The map drawing software is the place where you handle legibility and minimum distances. Your course setting software must not touch any rotation aspects.

**Comment by mlerjen** [comment]
@dg0yt: Just for clarification. What is the proper workflow to adjust existing maps to the relatively large changes of declination we see at the moment? For example it has  1.5 E and should be changed to 2.5E? Map and objects should be orientated towards the adjusted north, while map coordinates must be preserved (just tried map rotation but their the later was not the case).

**Comment by dg0yt** [comment]
@mlerjen I would update the declination field in the georeferencing dialog and apply the proposed map rotation. This should move move objects, templates and geographic references in the same way.

**Comment by mlerjen** [comment]
@dg0yt: Changing declination should not result in changed coordinates of the existing object but in the adjusted rotation of the objects oriented to north plus in a rotation of the map window (oriented towards "new north"). 
![image](https://user-images.githubusercontent.com/15611653/55331868-7bc65800-5494-11e9-9c11-666b141e7ac7.png)
fig.1. Result of changing declination (by 10 degrees) plus applying map rotation around geo-reference point. 100% is the base map, 65% the result (as geo-referenced image in background). What you can see is: geo-reference of the template (blue background) was not affected by the procedure. It still does accord to original map. Objects oriented to north got turned by the according angle. As far as good. It is just that the objects are no longer at their right position// no longer properly georeferenced.

**Comment by Zerbembasqwibo** [comment]
There is something here... But better put it in a new issue (or existing).

Before rotation of the map the GPX-file ends up like this:

![image](https://user-images.githubusercontent.com/16168975/55511940-22ac1f00-5662-11e9-9337-2b276a2e5620.png)

And after:

![image](https://user-images.githubusercontent.com/16168975/55511905-090ad780-5662-11e9-8631-40b10009b0fd.png)

**Comment by dg0yt** [comment]
> better put it in a new issue

Exactly. It is not about ideas.

**Comment by ollesmaps** [comment]
Quoting @lpechacek  "On a more personal note, after two years of following this project and attempting to contribute code I've got growing feeling that interested project contributors should gather somewhere for a face to face meeting to discuss past experiences, interests, standards for cooperation, future plans and building mutual trust. A multi-day orienteering event might be a good opportunity for such a meeting."

I agree. I can help organize such a meet up. WOC 2021 in Czech Rep. might be a good platform.  It is a little far in future though.

**Comment by ollesmaps** [comment]
Would you consider this an improvement (Android)?

**Toggle dash** in combination with **Point / Angle** button now adds or deletes dash points. 
I would prefer it only adds dash points.

(You can delete dash point using solo **Point / Angle** button and that is enough in my opinion)

**Comment by dg0yt** [comment]
@ollesmaps What you are describing is the effect of extra buttons in the mobile user interface which actually only simulate the press/hold of a modifier key (shift, control, alt, space) on a desktop keyboard. Certainly it would be an improvement to separate such features. It is just that the tools were not coded for that type of user interface. So it needs more work than it may seem.

**Comment by ollesmaps** [comment]
@dg0yt Thank you for explanation. I guess that is an issue for the future Android GUI overhaul. I let it be here.

**Comment by ghost** [comment]
> The map drawing software is the place where you handle ***legibility and minimum distances***.

Any dev progress on "Problem widget"?

- https://github.com/OpenOrienteering/mapper/issues/35

**Comment by ollesmaps** [comment]
There are new happy [converts](https://www.facebook.com/remmaps/photos/a.348636805150534/3121470721200448) :)

**Comment by ghost** [comment]
@dg0yt Has you plans to highlight *Mapper* development progress on 19th ICOM in Denmark? (Wednesday, July 8, 2020)

- https://onedrive.live.com/?cid=663580750d0c0bce&id=663580750D0C0BCE!19809&authkey=!ANDZzc98Fu1FG2k

**Comment by ghost** [comment]
### LGM 2020

It would be cool if someone could present *Mapper* development status at ***Libre Graphics Meeting***'`2020`, that will take place from *May 26th to Friday May 29th in Rennes, France* (*«CoVe»* overview would be interesting for LGM community)

- https://libregraphicsmeeting.org/2020/en/program.html
  - https://libregraphicsmeeting.org/2020/en/submission.html
- Deadline: ~~`February 17th`~~ extended to `March 1st` (to submit proposals)
  - https://twitter.com/LGMeeting/status/1231657768173461505

OR at least just drop few slides with short overview of *Mapper* for ***State of Libre Graphics***'`2020`, that would be highlighted as part of the opening presentation at *Libre Graphics Meeting*

- https://github.com/libregraphicsmeeting/state-of-lg-2020
- Deadline: `May 19th` (~ one week before *LGM'2020*)

**Comment by lpechacek** [comment]
> @ollesmaps: Contributors meetup ... WOC 2021 in Czech Rep. might be a good platform. It is a little far in future though.

FWIW, WOC '21 works for me.

**Comment by ollesmaps** [comment]
@lpechacek @dg0yt 
I approached the WOC2021 organizers and they will be happy to host a OOM meetup. Also the Czech Map Commission is welcoming and willing to help if needed. So if you think it is worth, I can devolop the idea further.

**Comment by ghost** [comment]
Programme for 19th International Conference on Orienteering Mapping (ICOM'2020) is available online:

- https://onedrive.live.com/?authkey=%21AEBjFaaRMy8Oqpk&cid=663580750D0C0BCE&id=663580750D0C0BCE%2145782&parId=663580750D0C0BCE%2119808&o=OneUp

TL;DR: There are no any presentations related to OpenOrienteering Mapper this year, only OCAD related stuff. Next ICOM would be in 2022.
